crates/bin/node/src/response_cache.rs

```diff
@@ -0,0 +1,82 @@
+use std::{
+    fmt::Debug,
+    time::{Duration, Instant},
+};
+
+use crate::errors;
+use dashmap::DashMap;
+use node::{MeltResponse, MintResponse, SwapResponse};
+
+pub trait ResponseCache<K, V> {
+    // Basic operations
+    fn get(&self, key: &K) -> Option<V>;
+    fn insert(&self, key: K, value: V) -> Result<(), errors::Error>;
+    fn remove(&self, key: &K) -> bool;
+
+    // For cleanup/shutdown
+    fn clear(&self);
+}
+
+#[derive(Debug)]
+pub struct InMemResponseCache<K, V>
+where
+    K: Eq + std::hash::Hash + Debug,
+    V: Clone,
+{
+    store: DashMap<K, (V, Instant)>,
+    ttl: Option<Duration>,
+}
+
+impl<K, V> InMemResponseCache<K, V>
+where
+    K: Eq + std::hash::Hash + Debug,
+    V: Clone,
+{
+    pub fn new(ttl: Option<Duration>) -> Self {
+        Self {
+            store: DashMap::new(),
+            ttl,
+        }
+    }
+}
+
+impl<K, V> ResponseCache<K, V> for InMemResponseCache<K, V>
+where
+    K: Eq + std::hash::Hash + Debug,
+    V: Clone,
+{
+    fn get(&self, key: &K) -> Option<V> {
+        let entry = self.store.get(key)?;
+        let (value, created_at) = &*entry;
+
+        if let Some(ttl) = self.ttl {
+            if created_at.elapsed() > ttl {
+                drop(entry);
+                self.store.remove(key);
+                return None;
+            }
+        }
```

Comment:
I get the idea, but in practice, it won't be enough
Once the wallet successfully receives its data, he won't be calling the route for this cached value anymore. So it will never be pruned that way; it will just expire and never be cleaned.

Reply:
We can do two things,
1. add a `prune` method to `InMemResponseCache` and trigger a tokio task to call it once every 15min like this. It will iter over all values and remove all the expired ones.
2. add an `Aknowledge(method, uuid)` to our gRPC server. The wallet will call this after he receives a response for mint_quote, min,t and melt operations. When called the node will remove those values from its cache.